parcels/collectionaos.py

```diff
@@ -0,0 +1,1130 @@
+from datetime import timedelta as delta
+from operator import attrgetter  # NOQA
+
+from ctypes import c_void_p
+
+import numpy as np
+
+from parcels.particlesets.collections import ParticleCollection
+from parcels.particlesets.iterators import BaseParticleAccessor
+from parcels.particlesets.iterators import BaseParticleCollectionIterator
+from parcels.particlesets.iterators import BaseParticleCollectionIterable
+from parcels.particle import ScipyParticle, JITParticle  # noqa
+from parcels.field import Field
+from parcels.tools.loggers import logger
+from parcels.tools.statuscodes import OperationCode
+
+try:
+    from mpi4py import MPI
+except:
+    MPI = None
+if MPI:
+    try:
+        from sklearn.cluster import KMeans
+    except:
+        raise EnvironmentError('sklearn needs to be available if MPI is installed. '
+                               'See http://oceanparcels.org/#parallel_install for more information')
+
+__all__ = ['ParticleCollectionAOS', 'ParticleCollectionIterableAOS', 'ParticleCollectionIteratorAOS']
+
+"""
+Author: Dr. Christian Kehl
+github relation: #913 (particleset_class_hierarchy)
+purpose: defines all the specific functions for a ParticleCollection, ParticleAccessor, ParticleSet etc. that relates
+         to a structure-of-array (SoA) data arrangement.
+"""
```

Comment:
well, it's usually standard for the files I create - but if you want, then I just remove that. Your system, your your decision.

Usually people do add that in code. Github is nice, but apart from internet-hype it doesn't count for anything because of pseudo-names etc. Also: if a contributor decides at some point to not be on github anymore (e.g. because one hosts one's personal subversioning system or any other reason), then his attribution gets eliminated. Fixing contributions in files makes their contribution permanent, recognizable and also legally binding. Overall, there's a reason why a majority of crucial software libraries include those things in the source files.